feat: implement admin panel using Administrate gem

Harendra-Singh-Rathore opened the pull request:
## Summary
Implemented an admin panel using the Administrate gem with custom banner field support for movie management. Added admin dashboards for Users, Movies, and Genres with proper field configurations, custom BannerField for Active Storage image handling, and enhanced the User model with password validation. This provides a complete administrative interface for managing the movie review application's data.

## Context
We needed a robust admin interface to manage the movie review application's data including users, movies, genres, ratings, and comments. By utilizing:

- Administrate gem for rapid admin panel generation
- Custom BannerField for handling movie banner images via Active Storage
- Enhanced dashboard configurations with proper field types and display options
- Password validation improvements for better security

This ensures administrators can efficiently manage all application data through a clean, intuitive interface while maintaining data integrity and security.

## Artifacts
- `Gemfile` - Added administrate, administrate-field-active_storage, faker, factory_bot_rails
- `app/controllers/admin/movies_controller.rb` - Custom controller with scoped resource ordering
- `app/dashboards/movie_dashboard.rb` - Movie admin dashboard with banner field support
- `app/dashboards/user_dashboard.rb` - User admin dashboard with password fields
- `app/dashboards/genre_dashboard.rb` - Genre admin dashboard configuration
- `app/fields/banner_field.rb` - Custom field for Active Storage banner images
- `app/views/fields/banner_field/` - Custom view templates for banner display
- `app/views/admin/` - Admin interface views
- `app/models/user.rb` - Enhanced password validation

## Impact
- [ ] Low (UI / small change)  
- [x] Medium (DB query, new model, background job)  
- [ ] High (affects login, payments, or core flows)  

### Screenshots
<img width="1436" height="794" alt="image" src="https://github.com/user-attachments/assets/6ec27c63-fd2a-4fa6-b51f-9a225d31c660" />


<img width="1438" height="761" alt="image" src="https://github.com/user-attachments/assets/4b6ea29e-47ce-4032-86dd-769e63e770e2" />


### Related Issue
Closes #5

Harendra-Singh-Rathore committed:
feat: add admin panel with full CRUD operations

Harendra-Singh-Rathore committed:
feat: implement admin panel with Administrate gem and custom banner field

- Add Administrate gem and administrate-field-active_storage for admin dashboard
- Create custom BannerField for displaying movie banners in admin interface
- Configure admin dashboards for movies, users, and genres with proper field types
- Add admin views and controllers for managing movies, users, and other resources
- Implement password validation with complexity requirements in User model

mandalorian99 commented on app/assets/stylesheets/application.css:
```diff
@@ -1,15 +0,0 @@
-/*
- * This is a manifest file that'll be compiled into application.css, which will include all the files
- * listed below.
- *
- * Any CSS (and SCSS, if configured) file within this directory, lib/assets/stylesheets, or any plugin's
- * vendor/assets/stylesheets directory can be referenced here using a relative path.
- *
- * You're free to add application-wide styles to this file and they'll appear at the bottom of the
- * compiled file so the styles you add here take precedence over styles defined in any other CSS
- * files in this directory. Styles in this file should be added after the last require_* statement.
- * It is generally better to create a new file per style scope.
- *
- *= require_tree .
- *= require_self
- */
```
@Harendra-Singh-Rathore is it deleted  accidently  or for some reason?

Harendra-Singh-Rathore replied:
No, it wasn't deleted accidentally. I moved the admin styles from application.css to a separate admin.css file for better organization and separation of concerns. The application.css is now clean and ready for general app styles.

mandalorian99 commented on app/controllers/admin_sessions_controller.rb:
```diff
@@ -0,0 +1,47 @@
+
+  def create
+
+    email = params[:email]&.strip&.downcase
+    password = params[:password]
+
+    #validate email
+    if email.blank?
+      flash.now[:alert] = "Email is required."
+      render :new
+      return
+    end
+
+    unless email.match?(/\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i)
+      flash.now[:alert] = "Please enter a valid email address."
+      render :new and return
+    end
+
+    #validate password
+    if password.blank?
+      flash.now[:alert] = "Password is required."
+      render :new
+      return
+    end
+
+    # Use the processed email variable, not params[:email]
+    admin = AdminUser.find_by(email: email)
+
+    if admin&.authenticate(password) 
+      session[:admin_user_id] = admin.id
+      redirect_to admin_root_path, notice: "Logged in successfully."
+    else
+      flash.now[:alert] = "Invalid email or password."
```
this logic could be extracted as a module `authenticable` or concern which can be imported into session controller .

Harendra-Singh-Rathore replied:
Good point! You're right, the authentication logic could be extracted into a concern or module for better reusability and separation of concerns. I'll refactor this into an Authenticatable concern that can be shared across session controllers.

Harendra-Singh-Rathore replied:
updated with concern

mandalorian99 commented on test/factories/admin_users.rb:
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :admin_user do
+    name { "MyString" }
+    email { "MyString" }
+    password_digest { "MyString" }
+  end
```
what will be the default email and password generated for admin ?

Harendra-Singh-Rathore replied:
Default Admin Credentials:
Email: admin@example.com
Password: password

Note: The factory file was auto-generated by FactoryBot when creating the AdminUser model, but we're using fixtures instead. The factory file can be deleted since it's redundant.

mandalorian99 reviewed (changes requested):
@Harendra-Singh-Rathore  nothing blocking , but check comments .

Harendra-Singh-Rathore committed:
feat(admin,ui,auth): add admin navigation, styles, and auth concern

- Add  and integrate with admin sessions
- Create  for admin navigation
- Add Administrate styles under
- Update
- Remove obsolete
